### ISSUE TITLE
Next.js error Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.

### ISSUE THREAD

**Issue by georgec-travellocal** [kevinfarrugia/react-glider#143]
For some reason when I use <Glider/> component in a package which I then do yarn link to another project, Next.js thinks Glider is an object not a react component.

Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.

The same occurs with `react-slick`

Any idea why this is?

I tried dynamic import with ssr false but it still fails after the client side is ready.

**Comment by kevinfarrugia** [comment]
Hi @georgec-travellocal .

I had created a [demo repository](https://github.com/kevinfarrugia/next-react-glider) using NextJS without any issue. 

Can you take a look and see if there is anything different in the implementation? Heads up I might be using an older version of Next.

**Comment by georgec-travellocal** [comment]
Yeah the error didn't exist with react-slick in next 12 but does with next
13.

**Comment by georgec-travellocal** [comment]
Also, the difference is we are importing the component which contains
<Glider> in a Next.js app as a package. The storybook for the package it
works fine, it's only after importing it as a package does it get those
errors.

**Comment by kevinfarrugia** [comment]
Will update to Next 13 and take a look.

**Comment by georgec-travellocal** [comment]
The code I'm using for the glider is like this:

import Glider  from 'react-glider';
import 'glider-js/glider.min.css';
...
const gliderRef = useRef<any>(null);
...
<Glider
  draggable={swipeToSlide}
  slidesToShow={slideCount}
  slidesToScroll={1}
  ref={gliderRef}
  onAnimated={evt => {
    const nextSlideIndex = evt.detail.value
    setIndex(nextSlideIndex);
    beforeChange?.(currentIndex, nextSlideIndex);
  }}
>
  {children}
</Glider>

It works fine in the storybook of the ui package, but when we use these
glider components in the package from the next.js website, it throws that
error.

**Comment by coreydunkin** [comment]
Unsure if this is the same issue as yours @georgec-travellocal but had a similar issue where react-glider would work inside storybook but wouldn't work when exposed to a Nextjs 13 application. I kept getting the error "Cannot set properties of undefined (setting 'Glider')"

The workaround was to import the react-glider library inside a useEffect like so:

`  const [Glider, setGlider] = useState<any>(null);
  const gliderRef = useRef();

  useEffect(() => {
    import('react-glider').then((GliderModule) => {
      setGlider(() => GliderModule.default);
    });
  }, []);`
  
  This removes the error after storybook is bundled and exposed to our NextJS app.
